src/launchoptions.rs

```diff
@@ -0,0 +1,211 @@
+            .captures(binding)
+            .context("No capture groups found -  no apps list present in localconfig.vdf.")?;
+
+        let mut matched_app_block: Option<String> = None;
+        let open_match = format!("\t\t\t\t\t\"{}\"", target_app);
+
+        // caps should be an iter of one element.
+        caps.iter().for_each(|mat_opt: Option<Match<'_>>| {
+            if let Some(mat) = mat_opt {
+                let latter_portion = mat
+                    .as_str()
+                    .find(&open_match)
+                    .context(
+                        "Could not find specified app in localconfig.vdf for the current user.",
+                    )
+                    .unwrap();
+                let latter = mat.as_str().substring(latter_portion, mat.end());
+                let first_app_close = latter
+                    .find("\n\t\t\t\t\t}")
+                    .context("Failed to find object closing statement for the matched app block.")
+                    .unwrap();
+                let app_match = latter.substring(0, first_app_close);
+
+                if matched_app_block.is_none() {
+                    matched_app_block = Some(app_match.to_string());
+                }
+            }
+        });
+
+        let launch_options_regex =
+            Regex::new(r#"\t{6}"LaunchOptions"\t{2}"([(\-\w)\s]*)""#).unwrap();
+
+        Ok(LaunchOptionsV2 {
+            local_config: config_path,
+            target_app,
+            launch_args_regex: launch_options_regex,
+            app_data: matched_app_block,
+            new_app_data: None,
+            required_options: required_opts,
+        })
+    }
+
+    pub fn check_missing_args(&self) -> Result<Vec<&str>, anyhow::Error> {
+        let span = tracing::span!(Level::INFO, "MissingLaunchOptions");
+        let _enter = span.enter();
+        tracing::info!("Checking for missing launch arguments in specified app...");
+        let mut missing_args: Vec<&str> = Vec::new();
+        let data_ref = match self.new_app_data {
+            Some(_) => &self.new_app_data,
+            None => &self.app_data,
+        };
+        let app_data = data_ref.clone().context("No data currently stored.")?;
+        let current_args = self
+            .launch_args_regex
+            .find(&app_data)
+            .context("Failed to find launch args object.")?;
+
+        let mat_str = current_args.as_str();
+        self.required_options.iter().for_each(|opt| {
+            if !mat_str.contains(opt) {
+                tracing::debug!("Launch Arguments: Missing argument identified -> {}", opt);
+                missing_args.insert(0, opt);
+            }
+        });
+
+        if !missing_args.is_empty() {
+            tracing::warn!("Missing arguments found in launch options for specified app...");
+            tracing::warn!("Missing arguments: {:?}", missing_args);
+        } else {
+            tracing::info!("All required launch arguments are present!");
+        }
```

Comment:
These types of logs should really be done where the result is handled instead of deep inside the function. Since you log an error back in `main.rs` when there are options missing, that means the logs here are going to print as well and there ends up being a bunch of duplicated information in the log which doesn't add anything.